### PR TITLE
build: add nix flake for linting + devShell + workflow

### PR DESCRIPTION
**Pull request by mrcjkb**
I noticed there's a stylua run in the makefile, but there's no `.stylua.toml`.

- [x] That should probably be added before this gets merged.

This adds a Nix flake with:

- A devShell with things like `lua-language-server`, `luacheck`, `stylua`, ...
- `checks` that run the linters (I also included `editorconfig-checker`, but I'm not sure if we want that?).
- A GitHub workflow that runs the checks on push and PR.

If we decide to merge this, I would like to follow-up with a `check` that runs `busted` tests using [`neorocksTest`](https://github.com/nvim-neorocks/neorocks#in-a-nix-flake).

### CODE REVIEW

**Comment by mrcjkb** [comment]
Note: When run locally, the `stylua` check fails.

**Comment by mrcjkb** [comment]
Oh wait - there is a `stylua.toml` (not hidden) :thinking: I wonder why it's failing...

**Comment by vhyrro** [comment]
It's likely because you're not compiling stylua with 5.2 support, which is required for parsing of `goto` and some other stuff. With cargo it's as simple as `cargo install stylua --features lua52` but unsure how it works in nix.

**Comment by vhyrro** [comment]
Also a stylua.toml is also considered standard and it picks it up on my end with no problems :thinking:

**Comment by mrcjkb** [comment]
Let's see if the fix gets merged on the Nix side: https://github.com/cachix/pre-commit-hooks.nix/pull/346
The stylua behaviour is intended.

File: nix/plugin-overlay.nix [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  name,
+  self,
+}: final: prev: {
+  vimPlugins.rocks-nvim = final.pkgs.vimUtils.buildVimPluginFrom2Nix {
```
Comment (teto) [inline nix/plugin-overlay.nix:5]: careful, in latest nixpkgs, it's now `buildVimPlugin` ;)

**Review by teto — APPROVED** [review]

File: nix/plugin-overlay.nix (hunk 1 above)
Comment (mrcjkb) [reply nix/plugin-overlay.nix:5]: Hmm, maybe we should use `buildLuarocksPackage` instead.
It's gonna need toml-edit and some other dependencies.

**Pull request merged**